Packs/GuardiCore/Integrations/GuardiCoreV2/GuardiCoreV2.py

```diff
@@ -0,0 +1,521 @@
+
+    hr = filter_human_readable(result, human_columns=INCIDENT_COLUMNS)
+
+    # Make the values more readable
+    hr['start_time'] = timestamp_to_datestring(hr['start_time'])
+    hr['end_time'] = timestamp_to_datestring(hr['end_time'])
+
+    md = tableToMarkdown(f'{INTEGRATION_NAME} - Incident: {incident_id}', hr,
+                         headers=INCIDENT_COLUMNS)
+
+    return CommandResults(
+        outputs_prefix=f'{INTEGRATION_CONTEXT_NAME}.Incident',
+        outputs_key_field=incident_id,
+        readable_output=md,
+        raw_response=result,
+        outputs=result,
+    )
+
+
+def get_assets(client: Client, args: Dict[str, Any]) -> List[CommandResults]:
+    ip_address = args.get('ip_address', None)
+    name = args.get('name', None)
+    asset_id = args.get('asset_id', None)
+    limit = int(args.get('limit', 50))
+    offset = int(args.get('offset', 0))
+
+    if not ip_address and not name and not asset_id:
+        raise DemistoException(
+            f"{INTEGRATION_NAME} - Endpoint search must have ip, name or asset_id defined.")
+
+    params = {
+        "asset_id": asset_id,
+        "limit": limit,
+        "offset": offset
+    }
+
+    if ip_address:
+        params['search'] = ip_address
+    elif name:
+        params['search'] = name
+
+    results = client.get_assets(params)
+
+    endpoints = []
+    for res in results.get("objects"):
+        hostname = res.get("guest_agent_details", {}).get("hostname", "")
+
+        res = filter_human_readable(res, human_columns=ASSET_COLUMNS)
+        md = tableToMarkdown(f'{INTEGRATION_NAME} - Asset: {hostname}',
+                             res, headers=ASSET_COLUMNS)
+
+        endpoints.append(CommandResults(
+            readable_output=md,
+            outputs_prefix=f'{INTEGRATION_CONTEXT_NAME}.Endpoint',
+            outputs_key_field="_id",
+            raw_response=res,
+            outputs=res,
+        ))
+
+    return endpoints
+
+
+def endpoint_command(client: Client, args: Dict[str, Any]) -> \
+        List[CommandResults]:
+    id = args.get("id", None)
+    ip_address = args.get("ip", None)
+    hostname = args.get("hostname", None)
+    if not id and not ip_address and not hostname:
+        raise DemistoException(
+            f'{INTEGRATION_NAME} - In order to run this command, please provide valid id, ip or hostname')
+
+    params = {}
+    if id:
+        params['search'] = id
+    elif ip_address:
+        params['search'] = ip_address
+    elif hostname:
+        params['search'] = hostname
+
+    results = client.get_assets(params)
+
+    endpoints = []
+    for res in results.get("objects"):
+        hostname = res.get("guest_agent_details", {}).get("hostname", "")
+        endpoint = Common.Endpoint(
+            hostname=hostname,
+            id=res.get("_id"),
+            os_version=res.get("guest_agent_details", {}).get(
+                "os_details", {}).get("os_display_name"),
+            ip_address=res.get("ip_addresses", []),
```

Comment:
the output key of IPAddress should be divided to ipv4, ipv6. if needed, edit it in the context standard.

Reply:
I don't understand where I specify it via `Common.Endpoint` I only see `ip_addresss` with no distinction for ipv4/ipv6

Reply:
Indeed, which is why I urge you to add, e.g: `self.ipv6_address = ipv6_address` property, can you set 5 with @David-BMS to define that?

Reply:
you can open an enhancement to support ipv6 and leave only the ipv4 address for now in the code, if the design keeps this pr from merging

Reply:
using the first one and opening an issue